packages/typescript/ai/tests/tools-per-model-type-safety.test.ts

```
@@ -0,0 +1,82 @@
+/**
+ * Type-safety tests for TextActivityOptions['tools'] gating.
+ *
+ * Mirrors the pattern in image-per-model-type-safety.test.ts — uses
+ * `@ts-expect-error` to assert compile-time rejections and `expectTypeOf`
+ * for positive inference checks.
+ */
+import { describe, it, expectTypeOf } from 'vitest'
```

ESLint

[error] Member 'expectTypeOf' of the import declaration should be sorted alphabetically.

(sort-imports)